tiled/client/base.py

```diff
@@ -370,6 +380,99 @@ def update_metadata(self, metadata=None, specs=None):
+                `update_metadata()` implements a similar approach to json merge
+                patch (RFC 7386 https://datatracker.ietf.org/doc/html/rfc7386).
+                However, None (or null) does not take a special meaning,
+                although `tiled.client.metadata_update.DELETE_KEY` does!
+        specs : List[str], optional
+            List of names that are used to label that the data and/or metadata
+            conform to some named standard specification.
+
+        """
+
+        if metadata is None:
+            metadata = {}
+
+        patch = jsonpatch.JsonPatch.from_diff(
+            dict(self.metadata),
+            apply_update_patch(self.metadata_copy(), metadata),
+            dumps=orjson.dumps,
+        ).patch
+
+        self.patch_metadata(patch=patch, specs=specs)
+
+    def patch_metadata(self, patch=None, specs=None):
+        """
+        EXPERIMENTAL: Patch metadata.
+
+        This is subject to change or removal without notice
+
+        Parameters
+        ----------
+        patch : List[dict], optional
+            JSON-serializable RFC 6902 patch to be applied to metadata
+            (See https://datatracker.ietf.org/doc/html/rfc6902)
+        specs : List[str], optional
+            List of names that are used to label that the data and/or metadata
+            conform to some named standard specification.
+        """
+
+        self._cached_len = None
+
+        if specs is None:
+            normalized_specs = None
+        else:
+            normalized_specs = []
+            for spec in specs:
+                if isinstance(spec, str):
+                    spec = Spec(spec)
+                normalized_specs.append(asdict(spec))
+
+        data = {
+            "patch": patch,
+            "specs": normalized_specs,
+        }
+
+        content = handle_error(
+            self.context.http_client.patch(
+                self.item["links"]["self"],
+                content=safe_json_dump(data),
+                headers={"Content-Type": "application/json-patch+json"},
+            )
+        ).json()
+
+        if patch is not None:
+            if "metadata" in content:
+                # Metadata was accepted and modified by the specs validator on the server side.
+                # It is updated locally using the new version.
+                self._item["attributes"]["metadata"] = content["metadata"]
+            else:
+                # Metadata was accepted as it is by the server.
+                # It is updated locally with the version submitted buy the client.
+                self._item["attributes"]["metadata"] = jsonpatch.apply_patch(
+                    dict(self.metadata), patch
+                )
+
+        if specs is not None:
+            self._item["attributes"]["specs"] = normalized_specs
+
+    def replace_metadata(self, metadata=None, specs=None):
+        """
+        EXPERIMENTAL: Replace metadata.
+
+        This is subject to change or removal without notice
+
```

Comment:
Do we need to make any changes to `metadata_revisions`, here or elsewhere? Or is it unrelated?

Reply:
Good point. When you call `update_metadata` it creates a revision before updating the metadata/specs. https://github.com/bluesky/tiled/blob/3b07e5f5add4d5ad2c5096a3694bee63d1a7308a/tiled/catalog/adapter.py#L836-L844

`metadata_revisions` always queries latest revisions from the server. So I think we should be fine there.